python3-requirements.yaml

```diff
@@ -0,0 +1,68 @@
+# Generated with flatpak-pip-generator --yaml --checker-data --requirements-file requirements.txt
+name: python3-requirements
+build-commands: []
+buildsystem: simple
+modules:
+  - name: python3-numpy
+    cleanup: [ /bin ]
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} "numpy~=1.22.3" --no-build-isolation
+    sources:
+      - &id001
+        type: file
+        url: https://files.pythonhosted.org/packages/b0/f4/d67c8c39efe3c45dfd32bb2a3fc49cbbe5496e575cc42b8bac60fe7b6701/numpy-1.22.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+        sha256: a911e317e8c826ea632205e63ed8507e0dc877dcdc49744584dfc363df9ca08c
+        only-arches: ["x86_64"]
+      - &id002
+        type: file
+        url: https://files.pythonhosted.org/packages/91/18/7949972b5254f88f040f53dc6f019fcb21597d6f37f7446d71019acece03/numpy-1.22.4-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
+        sha256: 43a8ca7391b626b4c4fe20aefe79fec683279e31e7c79716863b4b25021e0e74
+        only-arches: ["aarch64"]
+  - name: python3-pint
+    cleanup: [ /bin ]
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} "pint~=0.19.2" --no-build-isolation
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/eb/59/73fd3810d94f2c87aacc79f4e578e1d81df6f0f2800b80b7d3c4fb4e3a2d/Pint-0.19.2.tar.gz
+        sha256: e1d4989ff510b378dad64f91711e7bdabe5ca78d75b06a18569ac454678c4baf
+  - name: python3-psutil
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} "psutil~=5.9.0" --no-build-isolation
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/3d/7d/d05864a69e452f003c0d77e728e155a89a2a26b09e64860ddd70ad64fb26/psutil-5.9.4.tar.gz
+        sha256: 3d7f9739eb435d4b1338944abe23f49584bde5395f27487d2ee25ad9a8774a62
+  - name: python3-pyqtgraph
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} "pyqtgraph~=0.12.4" --no-build-isolation
+    sources:
+      - *id001
+      - *id002
```

Comment:
I don't think you need this. the package will already be installed.

Also I doubt this is the output of `atpak-pip-generator` which ideally you wouldn't want to touch.

Reply:
There is no other way than to edit the output of `flatpak-pip-generator` because it doesn't support wheels and also refuses to work with anything that depends on PyQt.

Reply:
But we also depend on pyqtgraph which in turn depends on PyQt, but pyqtgraph is not part of the base app... Also using wheels is almost inevitable, because compiling numpy and scipy from source is not easy and takes a long time.

Reply:
A few package I maintain build numpy from source.

Reply:
Okay, if you insist I can build numpy from source even though that will introduce dependencies that we don't need otherwise (Cython). Scipy will be a lot more difficult though. Scipy has several heavy dependencies, like BLAS+LAPACK (which we luckily do have) and pybind11.

Reply:
Please also note that a clean build on x86_64 already takes over 1 hour.

Reply:
Btw, this is the output of `flatpak-pip-generator` (patched to allow `pyqtgraph`) for a file with the relevant requirements:
```yaml
# Generated with flatpak-pip-generator --yaml -r requirements.txt
build-commands: []
buildsystem: simple
modules:
  - name: python3-numpy
    buildsystem: simple
    build-commands:
      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
        --prefix=${FLATPAK_DEST} "numpy~=1.22.3" --no-build-isolation
    sources:
      - &id001
        type: file
        url: https://files.pythonhosted.org/packages/f6/d8/ab692a75f584d13c6542c3994f75def5bce52ded9399f52e230fe402819d/numpy-1.22.4.zip
        sha256: 425b390e4619f58d8526b3dcf656dde069133ae5c240229821f01b5f44ea07af
  - name: python3-pint
    buildsystem: simple
    build-commands:
      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
        --prefix=${FLATPAK_DEST} "pint~=0.19.2" --no-build-isolation
    sources:
      - type: file
        url: https://files.pythonhosted.org/packages/eb/59/73fd3810d94f2c87aacc79f4e578e1d81df6f0f2800b80b7d3c4fb4e3a2d/Pint-0.19.2.tar.gz
        sha256: e1d4989ff510b378dad64f91711e7bdabe5ca78d75b06a18569ac454678c4baf
  - name: python3-pyqtgraph
    buildsystem: simple
    build-commands:
      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
        --prefix=${FLATPAK_DEST} "pyqtgraph~=0.12.4" --no-build-isolation
    sources:
      - *id001
      - type: file
        url: https://files.pythonhosted.org/packages/83/44/d84034102ff3888ca33a7e7840b0a2831fbf5b9075918e5a88be1c733397/pyqtgraph-0.12.4-py3-none-any.whl
        sha256: d98c6e26e9eda6d1d1bba92d6df381ef77fca18b701f9e136759ce31a17c6217
  - name: python3-scipy
    buildsystem: simple
    build-commands:
      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
        --prefix=${FLATPAK_DEST} "scipy~=1.8.0" --no-build-isolation
    sources:
      - *id001
      - type: file
        url: https://files.pythonhosted.org/packages/26/b5/9330f004b9a3b2b6a31f59f46f1617ce9ca15c0e7fe64288c20385a05c9d/scipy-1.8.1.tar.gz
        sha256: 9e3fb1b0e896f14a85aa9a28d5f755daaeeb54c897b746df7a55ccb02b340f33
name: python3-requirements
```
No offense, but this is completely useless because it doesn't capture *any* of the dependencies of numpy and scipy.

Reply:
I've moved from flatpak-pip-generator to [req2flatpak](https://github.com/johannesjh/req2flatpak).